packages/s3-store/index.ts

```diff
@@ -205,18 +198,15 @@ export class S3Store extends DataStore {
   }
 
   private async deleteIncompletePart(id: string): Promise<void> {
-    await this.client
-      .deleteObject({
-        Bucket: this.bucket,
-        Key: id,
-      })
-      .promise()
+    await this.client.deleteObject({
+      Bucket: this.bucket,
+      Key: id,
+    })
   }
 
-  private async prependIncompletePart(path: string, buffer: Buffer): Promise<void> {
-    const part = await fsProm.readFile(path, 'utf8')
-    buffer.write(part, buffer.length - 1)
-    await fsProm.writeFile(path, buffer)
+  private async prependIncompletePart(path: string, part: Uint8Array): Promise<void> {
```

Comment:
This function confuses me a bit. The arguments' names are not easy to understand. I assume `path` points to the file with the data that we just received from the client and `part` is the incomplete part that we fetched from S3. If so, maybe the names could be improved or a comment could be added.

In addition, I think the order in `Buffer.concat` is wrong. Currently it appends the incomplete part, but we want to prepend it.

Reply:
I'll try to clear it up. However, concat should be working as expected as the test passes:

https://github.com/tus/tus-node-server/blob/ca1bb7a3c87701dbfc70d16b867186daf89cd4fd/packages/s3-store/test.ts#L36-L42

Reply:
Update: the test also passes if I flip the argument. Something is not right.

Reply:
I used `assert()` instead of `assert.strictEqual()`...

Reply:
Oh, good one :D

AFAIK, the original logic before this PR and now after the PR are correct, right?